src/reducers/books.js

```diff
@@ -0,0 +1,20 @@
+import { CREATE_BOOK, REMOVE_BOOK } from '../actions/index';
+
+const booksReducer = (state = [], action) => {
+  switch (action.type) {
+    case CREATE_BOOK:
+      return {
+        ...state,
+        newbook: action.payload,
+      };
+    case REMOVE_BOOK:
+      return {
+        ...state,
+        REMOVE_BOOK: true,
+      };
+    default:
+      return state;
+  }
+};
```

Comment:
**[Required]** The bookReducer function is not correct. I think you are mismatching actions and reducer. For example, We get {[existingbookArray, newbook: book]} when CREATE_BOOK is dispatched. In the REMOVE_BOOK you are not passing **id** the book that needs to be deleted. How does it know which book to remove? We can pass an id as a payload and filter the book from bookArray. [Reference](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/ReducxReducers/CreateAndRemoveReducers.png)